drivers/scheduler/k8s/k8s.go

```diff
@@ -5047,12 +5048,191 @@ func (k *K8s) CreateCsiSnapsForVolumes(ctx *scheduler.Context, snapClass string)
+	cmdArgs := []string{"exec", "-it", restoredPod.Name, "-n", namespace, "--", "bin/cat", "/mnt/volume1/aaaa.txt"}
+	command := exec.Command("kubectl", cmdArgs...)
+	fileConent, err := command.Output()
+	if !strings.Contains(string(fileConent), dirtyData) {
+		return fmt.Errorf("restored volume does NOT contain data from original volume")
+
+	}
+	logrus.Info("Validation complete")
+	return nil
+}
+
+// MakePod Returns a pod definition based on the namespace. The pod references the PVC's
+// name.  A slice of BASH commands can be supplied as args to be run by the pod
+func MakePod(ns string, pvclaims []*v1.PersistentVolumeClaim, command string, privileged bool) *v1.Pod {
+	var cmd = "while true; do sleep 1; done"
+	if 0 < len(command) {
+		cmd = command + " && " + cmd
+	}
+
+	podSpec := &v1.Pod{
+		TypeMeta: metav1.TypeMeta{
+			Kind:       "Pod",
+			APIVersion: "testMountSnapshotToPod",
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			GenerateName: "testpod-",
+			Namespace:    ns,
+		},
+		Spec: v1.PodSpec{
+			Containers: []v1.Container{
+				{
+					Name:            "test-pod",
+					Image:           "alpine:3.10",
+					Command:         []string{"/bin/sh", "-c"},
+					Args:            []string{cmd},
+					SecurityContext: &v1.SecurityContext{Privileged: &privileged},
+				},
+			},
+			RestartPolicy: v1.RestartPolicyOnFailure,
+		},
+	}
+	var volumeMounts = make([]v1.VolumeMount, 0)
+	var volumeDevices = make([]v1.VolumeDevice, 0)
+	var volumes = make([]v1.Volume, len(pvclaims))
+	for index, pvclaim := range pvclaims {
+		if pvclaim.Spec.VolumeMode != nil {
+			fmt.Printf("mount pvc %s as %s\n", pvclaim.Name, *pvclaim.Spec.VolumeMode)
+		} else {
+			fmt.Printf("mount pvc %s, volume mode is nil\n", pvclaim.Name)
+		}
+		volumename := fmt.Sprintf("volume%v", index+1)
+
+		if pvclaim.Spec.VolumeMode == nil || *pvclaim.Spec.VolumeMode == v1.PersistentVolumeFilesystem {
+			volumeMounts = append(volumeMounts, v1.VolumeMount{Name: volumename, MountPath: "/mnt/" + volumename})
+			volumes[index] = v1.Volume{Name: volumename, VolumeSource: v1.VolumeSource{PersistentVolumeClaim: &v1.PersistentVolumeClaimVolumeSource{ClaimName: pvclaim.Name, ReadOnly: false}}}
+		} else {
+			// Raw block device
+			volumeDevices = append(volumeDevices, v1.VolumeDevice{Name: volumename, DevicePath: fmt.Sprintf("/dev/xvda%d", index)})
+			volumes[index] = v1.Volume{Name: volumename, VolumeSource: v1.VolumeSource{PersistentVolumeClaim: &v1.PersistentVolumeClaimVolumeSource{ClaimName: pvclaim.Name, ReadOnly: false}}}
+		}
+	}
+	podSpec.Spec.Containers[0].VolumeMounts = volumeMounts
+	podSpec.Spec.Containers[0].VolumeDevices = volumeDevices
+	podSpec.Spec.Volumes = volumes
+	return podSpec
+}
+
+// MakePVC takes namespace, name, source snapshot name and storageclass as parameter and returns a PersistentVolumeClaim spec for PVC creation
+func MakePVC(ns string, name string, sourceSnapshotName string, storageClass string) *v1.PersistentVolumeClaim {
```

Comment:
I suggest you have 3 functions:
```
func MakePVC(ns string, targetName string, storageClass string)
func MakePVCfromSnapshot(ns string, targetName string, sourceSnapshotName string, storageClass string)
func ClonePVCfromVolume(ns string, targetName string, sourceVolumeName string, storageClass string)
```

Reply:
sounds good, I will have the first 2 first 2 for these PR and will on ClonePVCfromVolume when I get to clone next Sprit.